namesys/namesys.go

```diff
@@ -112,12 +112,12 @@ func (ns *mpns) resolveOnceAsync(ctx context.Context, name string, options opts.
 	}
 
 	// Resolver selection:
-	// 1. if it is a multihash resolve through "ipns".
+	// 1. if it is a CID/multihash resolve through "ipns".
 	// 2. if it is a domain name, resolve through "dns"
 	// 3. otherwise resolve through the "proquint" resolver
 
 	var res resolver
-	if _, err := mh.FromB58String(key); err == nil {
+	if _, err := cid.Decode(key); err == nil {
```

Comment:
We need to decode this as a peer ID. This will break ed25519 IPNS keys (which we should probably have a test for).

https://pkg.go.dev/github.com/libp2p/go-libp2p-core@v0.3.1/peer?tab=doc#Decode

Reply:
Good catch.
I will switch to `peer` and add tests for missing PeerID "types" (ed25519, libp2p-key in cid, etc).